lib/lorekeeper/json_logger.rb

```diff
@@ -97,11 +98,19 @@ def inspect
 
     # Some instrumentation libraries pollute the stacktrace and create a large output which may
     # cause problems with certain logging backends.
-    # Hardcoring newrelic now here. In the future if this list grows, we may make it configurable.
+    # Hardcoring newrelic and active_support/callbacks now here.
+    # In the future if this list grows, we may make it configurable.
     def clean_backtrace(backtrace)
-      backtrace.reject do |line|
-        line.include?(BLACKLISTED_FINGERPRINT)
-      end
+      @backtrace_cleaner ? @backtrace_cleaner.clean(backtrace) : backtrace
+    end
+
+    def set_backtrace_cleaner
```

Comment:
@abhitrivedi This means activesupport (not rails) is not needed but you get extra functionality, sounds good?